internal/registry/validate.go

```diff
@@ -16,17 +16,24 @@ package registry
 
 import (
 	"fmt"
+	"io/ioutil"
 
 	apimanifests "github.com/operator-framework/api/pkg/manifests"
 	apivalidation "github.com/operator-framework/api/pkg/validation"
 	apierrors "github.com/operator-framework/api/pkg/validation/errors"
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
 	registrybundle "github.com/operator-framework/operator-registry/pkg/lib/bundle"
 	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
```

Comment:
```suggestion
	"sigs.k8s.io/yaml"
```

Reply:
Switching over to `gopkg.in/yaml.v2`. k8s-yaml interprets and appends the field of yaml struct while marshaling and unmarshaling, whereas gopkg-yaml uses the yaml annotation defined in the struct. Example - https://play.golang.org/p/uQ_J4oZBbCR

Reply:
I just replaced by the `"sigs.k8s.io/yaml"` and the tests worked successfully. Both libs help us work with yaml. However, we are trying to keep all project using `"sigs.k8s.io/yaml"` and remove any other module/dep which has the same goal. Ideally, how much fewer deps we have added to the project would be better to keep it maintained. Did you find any problem with ``"sigs.k8s.io/yaml"`` which justify you use the other lib?

Reply:
The unit tests do pass with `"sigs.k8s.io/yaml"`. However, it would fail in the scenario where we create bundle image using files already present on disk.
Explanation is as below:

To reproduce the scenario:
1. First create bundle files on disk with `operator-sdk bundle create` command and `--generate-only` flag.
2. Now create bundle image and push it to the registry (for eg. use the command: `operator-sdk bundle create  quay.io/example/memcached-operator:latest --directory path/to/manifests --package memcached-operator`)

Cause of error:
When we use `sigs.k8s.io/yaml`, the key in `annotations.yaml` file would be the name of the field in [`AnnotationMetadata`](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/generate.go#L35) struct. The `annotations.yaml` which would be rewritten after adding the sdk stamps, would be like this:
```
Annotations:
  operators.operatorframework.io.bundle.channel.default.v1: stable
  operators.operatorframework.io.bundle.channels.v1: stable
  operators.operatorframework.io.bundle.manifests.v1: manifests/
```

Now when during unmarshalling, ie [here](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/generate.go#L265), it would not be able to interpret this because the yaml has the key `Annotations ` not `annotations`. 

If we use `"gopkg.in/yaml.v2"`, the `annotations.yaml` which we re-write would take the marker value defined in yaml struct, `yaml:"annotations"` as it does in operator-registry and it would unmarshall properly. 

I have created a simple example to show the difference between both the packages in this playground: https://play.golang.org/p/uQ_J4oZBbCR

Reply:
See that your example worked if we replace the yaml for json

> Something int `yaml:"somethingGreat,omitempty"`
With
> Something int `json:"somethingGreat,omitempty"`

Then, see that also it will work with `yaml ` if we replace the `Unmarshal/Marchal` for `JSONToYAML/YAMLToJSON`. See: https://play.golang.org/p/D2guhgb7cvc

In SDK, we read/update/delete the files and use the lib to verify them. In this way, no matter the dep/lib that we use to manipulate it, all should work fine if at the end we have the same result.  Then, I understand that if we change the impl [here](https://github.com/operator-framework/operator-sdk/blob/c4d9a0ef121cee83e0549fce715ee200439b4aad/internal/registry/validate.go#L179-L203) to use the above it should work.  

However, if you still just facing problems with. could you please add a comment with a //todo: replace it for "sigs.k8s.io/yaml". Operator-registry defines the annotations with yaml format.

Reply:
Adding //todo: because of the dependency on operator-registry. OR defines annotations.

Reply:
This could likely be fixed by adding `json:"annotations"` to the struct tag in the operator-registry repo.

So it would look like: 

```go
type AnnotationMetadata struct {
	Annotations map[string]string `yaml:"annotations" json:"annotations"``
}

```

Reply:
Will continue with above and have an upstream PR in parallel. Ill have a follow-up PR to remove the TODO.